Allow registration of multiple providers and only abort, if none succeeded.

diff --git a/pam_saml.c b/pam_saml.c
index a91ddb9..a2025bd 100644
--- a/pam_saml.c
+++ b/pam_saml.c
@@ -278,6 +278,7 @@ pam_global_context_init(pamh, ac, av)
 		}
 	}
 
+	int num_of_idps = 0;
 	for (i = 0; i < ac; i++) {
 		const char *idp;
 
@@ -285,11 +286,10 @@ pam_global_context_init(pamh, ac, av)
 			continue;
 
 		if (access(idp, R_OK) != 0) {
-			error = PAM_OPEN_ERR;
 			syslog(LOG_ERR,
 			       "Unable to read IdP metadata file \"%s\"", 
 			       idp);
-			goto cleanup;
+			continue;
 		}
 
 		if (lasso_server_add_provider(gctx->lasso_server,
@@ -300,9 +300,13 @@ pam_global_context_init(pamh, ac, av)
 			       "Failed to load metadata from \"%s\"", idp);
 			goto cleanup;
 		}
-
+		num_of_idps++;
 		syslog(LOG_DEBUG, "Loaded metadata from \"%s\"", idp);
 	}
+	if (!num_of_idps) {
+		error = PAM_OPEN_ERR;
+		goto cleanup;
+	}
 
 	if ((gctx->uid_attr = strdup(uid_attr)) == NULL) {
 		error = PAM_SYSTEM_ERR;
